pulp_service/pulp_service/app/tasks/ansible_log_parser.py

```diff
@@ -0,0 +1,216 @@
+                        role,
+                        str(role_filter),
+                        log_url,
+                    )
+                    return []
+                ansible_version = match.group("ansible_ver")
+
+    # Use the async function to get file data
+    lines = await get_file_data(log_url)
+
+    for line in lines:
+        if (
+            line.startswith("TASK ")
+            or line.startswith("PLAY ")
+            or line.startswith("META ")
+        ):
+            # end of current task and possibly start of new task
+            if task_lines and task_has_fatal:
+                # Extract task name from the first task line
+                task_match = re.search(r"TASK\s\[(.*?)\]", task_lines[0])
+                if task_match:
+                    current_task = task_match.group(1)
+                # end task
+                error = {
+                    "Url": log_url,
+                    "Role": role,
+                    "Ansible Version": ansible_version,
+                    "Task": current_task,
+                    "Detail": copy.deepcopy(task_lines[3:]),
+                    "Task Path": task_path,
+                }
+                errors.append(error)
+            if line.startswith("TASK "):
+                task_lines = [line.strip()]
+            else:
+                task_lines = []
+            task_has_fatal = False
+            task_path = None
+        elif task_lines:
+            task_lines.append(line.strip())
+            if line.startswith("fatal:"):
+                task_has_fatal = True
+            elif line.startswith("failed:"):
+                task_has_fatal = True
+            elif line.startswith("task path:"):
+                task_path_match = re.search(r"task path: (.*)", line)
+                if task_path_match:
+                    task_path = task_path_match.group(1)
+            elif line.startswith("...ignoring"):
+                task_has_fatal = False
+        else:
+            match = re.search(r"\sfailed=(\d+)\s", line)
+            if match:
+                total_failed += int(match.group(1))
+
+    _logger.debug(
+        "Found [%d] errors and Ansible reported [%d] failures",
+        len(errors),
+        total_failed,
+    )
+    if total_failed == 0:
+        errors = []
+    for error in errors:
+        error["Fails expected"] = total_failed
+
+    return errors
+
+
+# Function to be used for dispatching the task
+def dispatch_ansible_log_analysis(log_url, role_filter):
+    """
+    Dispatch an async task to analyze an Ansible log file.
+    
+    Args:
+        log_url (str): URL of the Ansible log file
+        role_filter (list): List of roles to filter by
+
+    Returns:
+        Task: The dispatched task object
+    """
+    # Create resources list for domain isolation
+    exclusive_resources = []
+    
+    try:
+        domain = get_domain()
+        # Add as a exclusive_resources
+        exclusive_resources.append(domain)
```

Comment:
Does this need to be an exclusive resource? Every task that is dispatched has the domain as a shared resource by default. So if you specify that the domain should be an exclusive resource you are saying that it should block every other task in that domain.